grails-app/controllers/au/org/ala/spatial/service/LoginInterceptor.groovy

```diff
@@ -39,21 +39,40 @@ class LoginInterceptor {
             return true
         }
 
+        //Calculating the required permission.
+        def permissionLevel = null;
+        //Permission on method has the top priority
+        if ( method?.isAnnotationPresent(RequirePermission)) {
+            permissionLevel = RequirePermission
+        } else if (method?.isAnnotationPresent(RequireLogin)){
+            permissionLevel = RequireLogin
+        } else if (method?.isAnnotationPresent(RequireAdmin)){
+            permissionLevel = RequireAdmin
+        }
+
+        if (Objects.isNull(permissionLevel)) {
+            if ( controllerClass?.isAnnotationPresent(RequirePermission)) {
+                permissionLevel = RequirePermission
+            } else if (controllerClass?.isAnnotationPresent(RequireLogin)){
+                permissionLevel = RequireLogin
+            } else if (controllerClass?.isAnnotationPresent(RequireAdmin)){
+                permissionLevel = RequireAdmin
+            }
+        }
+
+        //Permission check
         def role  // if require a certain level of ROLE
-        if ( method?.isAnnotationPresent(RequirePermission) || controllerClass?.isAnnotationPresent(RequirePermission) ) {
+        if ( permissionLevel == RequirePermission ) {
             if (serviceAuthService.isLoggedIn() || serviceAuthService.hasValidApiKey()) {
                 return true
             } else {
                 return accessDenied(STATUS_UNAUTHORISED,'Forbidden, ApiKey or user login required!')
             }
-        }
-        // UserId is required
-        // May need to check role
-        if (method?.isAnnotationPresent(RequireAdmin) || controllerClass?.isAnnotationPresent(RequireAdmin)) {
+        } else if ( permissionLevel == RequireAdmin ) {
             if (serviceAuthService.hasValidApiKey())
                 return true
             role = grailsApplication.config.auth.admin_role //recommended: ROLE_ADMIN
```

Comment:
Please change `grailsApplication.config.auth.admin_role` to be `grailsApplication.config.getProperty('auth.admin_role', String, 'ROLE_ADMIN')` so it is type safe and has fallback default value. `grailsApplication.config.getProperty()` should always be used to read config vars...

Reply:
Done